### PR TITLE
React added to the projects and Readme updated

### PR DESCRIPTION
**Pull request by achhetr**
I have added react and react-dom dependency to the project. At present react dom operates in the root_path so that we can implement devise authentication without using react app. I have also updated the README document to make sure if anyone joins this project in the future, installation is easy to follow.

### CODE REVIEW

**Review by saramic — CHANGES_REQUESTED** [review]
Hi, not the style of review I generally like to give, my bad :(

I would rather review by asking questions why certain decisions were made, the rational, the thinking, how they will evolve over time. In this case I have just commented on the stuff we don't really need

have a go at doing some of the changes but over all GREAT 👍 keep it up

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
-  <body>
-    <p class="notice"><%= notice %></p>
-    <p class="alert"><%= alert %></p>
-
-    <%= yield %>
-  </body>
+	<body>
+		<%= render  "shared/flash_message" %>
+		<%= yield %>
+	</body>
```
[the remark this answers is not in the document]
Comment (achhetr) [reply app/views/layouts/application.html.erb:15]: I am not sure why it is changing to 4 characters as my vs code editor shows 2 spaces when I commit and push. This needs further investigation.

File: app/views/home/index.html.erb [hunk 2]
```diff
@@ -1 +1,2 @@
-Health Tracker
+<%# React App is working from the index.jsx%>
```
[the remark this answers is not in the document]
Comment (achhetr) [reply app/views/home/index.html.erb]: comments removed

File: app/javascript/packs/index.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Run this example by adding <%= javascript_pack_tag 'hello_react' %> to the head of your layout file,
```
[the remark this answers is not in the document]
Comment (achhetr) [reply app/javascript/packs/index.jsx]: comments removed

File: app/javascript/packs/components/App.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const App = () => {
+	return <div>React App</div>;
```
[the remark this answers is not in the document]
Comment (achhetr) [reply app/javascript/packs/components/App.js]: I have fixed this

File: app/controllers/home_controller.rb [hunk 5]
```diff
@@ -1,3 +1,5 @@
 class HomeController < ApplicationController
```
[the remark this answers is not in the document]
Comment (achhetr) [reply app/controllers/home_controller.rb]: I will need some more information on this

File: README.md [hunk 6]
```diff
@@ -1,24 +1,55 @@
...
+# Installation
 
-* Ruby version
+## Clone project
 
-* System dependencies
+You can clone the project using
 
-* Configuration
+`gh repo clone failure-driven/health-tracker`
```
[the remark this answers is not in the document]
Comment (achhetr) [reply README.md]: gh removed :) thanks

File: README.md [hunk 7]
```diff
@@ -1,24 +1,55 @@
...
+## System Requirements
 
-* Services (job queues, cache servers, search engines, etc.)
+* Ruby version **ruby-3.0.0**
 
-* Deployment instructions
+& Database **postgresql**
 
-* ...
+## Install Project Dependencies
```
[the remark this answers is not in the document]
Comment (achhetr) [reply README.md:29]: I have remove seeds info from the document, I will research more on make build

File: .eslintrc.js [hunk 8]
```diff
@@ -1,23 +1,21 @@
...
-    ecmaVersion: 12,
-    sourceType: 'module',
-  },
-  plugins: [
-    'react',
-  ],
-  rules: {
-  },
+	env: {
+		browser: true,
```
Comment (achhetr) [inline .eslintrc.js:3]: I am not sure why it is adding 4 character spaces, this needs more research

File: .eslintrc.js [hunk 9]
```diff
@@ -1,23 +1,21 @@
...
+		ecmaFeatures: {
+			jsx: true,
+		},
+		ecmaVersion: 12,
+		sourceType: 'module',
+	},
+	plugins: ['react'],
+	rules: {
+		'react/jsx-filename-extension': [1, { extensions: ['.js', '.jsx'] }],
+		indent: [2, 'tab'],
```
Comment (achhetr) [inline .eslintrc.js:18]: I found this rules to be effectiv

File: app/controllers/home_controller.rb [hunk 10]
```diff
@@ -1,3 +1,5 @@
...
   before_action :authenticate_user
+  def index
```
[the remark this answers is not in the document]
Comment (achhetr) [reply app/controllers/home_controller.rb]: removed

File: .eslintrc.js (hunk 8 above)
Comment (achhetr) [reply .eslintrc.js]: resolved

File: app/views/layouts/application.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (achhetr) [reply app/views/layouts/application.html.erb]: resolved

**Review by saramic — APPROVED** [review]

**Pull request merged**